plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/BuildPath.java

```diff
@@ -79,53 +73,66 @@ public static void addMavenLibraries(IProject project, List<Library> libraries,
   }
 
   /**
-   * @return the entries added to the classpath.
-   *     Does not include entries previously present in classpath.
+   * Returns the entries added to the classpath.
    */
-  static IClasspathEntry[] addLibraries(
-      IJavaProject javaProject, List<Library> libraries, IProgressMonitor monitor)
-          throws JavaModelException, CoreException {
-
-    // todo or we could grab the transitive dependency graph here if needed
-    // at this point we have the Java project so we could even filter
-    // entries already present in other libraries
+  public static IClasspathEntry[] addNativeLibrary(IJavaProject javaProject,
```

Comment:
I've been thinking about the interface for a while. Going forward, if we are going to make `BuildPath` not general in that it always and only adds a single master library, it'd be much clearer and less confusing if we rename this to `addNativeMasterLibrary` (or `addNativeMasterContainer` or `addNativeMasterLibraryContainer`). The Javadoc would also signify that this method returns either 1) a singleton `IClasspathEntry` array, whose only element is the master container; or 2) an empty array if the container already exists in the project.

In any case, I don't see anywhere its return value is used. We could make it `void`.

Perhaps we could create a new issue for this renaming and updating Javadocs.

Reply:
If we reach that point, I prefer the name `addNativeMasterContainer()` and rename the parameter `libraries` to `librariesToInclude`. Currently, it is kind of confusing in that the method sounds like it is adding a single "library" (`addNativeLibrary` - which is really the master container) but it is accepting multiple "libraries" (the parameter).

Reply:
Yes, at this point the return value is only used in tests so this probably should be void. I'm not sure how much those tests are testing now.
 addNativeLibraries, perhaps since it can add more than one. "ToInclude" doesn't seem necessary.
Let's not overthink this one as long as it works for now. I expect a lot more changes around all this code as we move to purely transitive dependencies, and any cleanup work we do here is likely to get reworked anyway.